### PR TITLE
Simplify specs

### PR DESCRIPTION
**Pull request by kevgo**
@charlierudolph @allewun 

Some specs create commits that aren't really necessary. I have gone through all feature specs and removed the ones that I think are not adding to the feature spec.

### CODE REVIEW

File: features/support/env.rb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 require 'active_support/all'
 require 'kappamaki'
 require 'open4'
+require 'pathname'
```
Comment (charlierudolph) [inline features/support/env.rb]: Why is this required now?

Comment (kevgo) [inline features/support/env.rb]: Sorry, this wasn't required here, but in another branch. Removed.

**Comment by kevgo** [comment]
It seems necessary for some specs. I have another PR that introduces it properly.

**Comment by kevgo** [comment]
bump

File: features/git-sync/all_branches/non_feature_branch_rebase_tracking_branch_conflict/last_branch/without_open_changes.feature [hunk 2]
```diff
@@ -1,27 +1,23 @@
 Feature: git sync --all: handling rebase conflicts between non-feature branch and its tracking branch without open changes
 
   Background:
-    Given I have branches named "production" and "qa"
```
Comment (charlierudolph) [inline features/git-sync/all_branches/non_feature_branch_rebase_tracking_branch_conflict/last_branch/without_open_changes.feature]: production is necessary to me here. This is showing proper skip behavior as its the last branch in a git sync all that hits a conflict

Comment (kevgo) [inline features/git-sync/all_branches/non_feature_branch_rebase_tracking_branch_conflict/last_branch/without_open_changes.feature]: Okay, I have restored this file

**Comment by kevgo** [comment]
All comments addressed

**Comment by charlierudolph** [comment]
:ship:
